From f08f0fa19625721778ff8685908e81e589df543e Mon Sep 17 00:00:00 2001
From: Rudi <r.ihle@s-t.de>
Date: Fri, 25 Dec 2015 09:06:46 +0100
Subject: [PATCH] video: mxc_edid: Cleanup and remove unused function

mxc_edid_var_to_vic() is exported but never used. Let's get rid of it. Also
streamline mxc_edid_mode_to_vic() a bit and make mxc_edid_fb_mode_is_equal()
static.

Signed-off-by: Rudi <r.ihle@s-t.de>
---
 drivers/video/mxc/mxc_edid.c | 27 +++------------------------
 include/video/mxc_edid.h     |  1 -
 2 files changed, 3 insertions(+), 25 deletions(-)

diff --git a/drivers/video/mxc/mxc_edid.c b/drivers/video/mxc/mxc_edid.c
index edfd6e8..ef8d261 100644
--- a/drivers/video/mxc/mxc_edid.c
+++ b/drivers/video/mxc/mxc_edid.c
@@ -216,7 +216,7 @@ EXPORT_SYMBOL(mxc_fb_mode_is_equal_res);
  * pixclock, since for many CEA modes, 2 frequencies are supported
  * e.g. 640x480 @ 60Hz or 59.94Hz
  */
-int mxc_edid_fb_mode_is_equal(bool use_aspect,
+static int mxc_edid_fb_mode_is_equal(bool use_aspect,
 			const struct fb_videomode *mode1,
 			const struct fb_videomode *mode2)
 {
@@ -741,24 +741,6 @@ static int mxc_edid_readsegblk(struct i2c_adapter *adp, unsigned short addr,
 	return ret;
 }
 
-int mxc_edid_var_to_vic(struct fb_var_screeninfo *var)
-{
-	int i;
-	struct fb_videomode m;
-
-	for (i = 0; i < ARRAY_SIZE(mxc_cea_mode); i++) {
-		fb_var_to_videomode(&m, var);
-		if (mxc_edid_fb_mode_is_equal(false, &m, &mxc_cea_mode[i]))
-			break;
-	}
-
-	if (i == ARRAY_SIZE(mxc_cea_mode))
-		return 0;
-
-	return i;
-}
-EXPORT_SYMBOL(mxc_edid_var_to_vic);
-
 int mxc_edid_mode_to_vic(const struct fb_videomode *mode)
 {
 	int i;
@@ -766,13 +748,10 @@ int mxc_edid_mode_to_vic(const struct fb_videomode *mode)
 
 	for (i = 0; i < ARRAY_SIZE(mxc_cea_mode); i++) {
 		if (mxc_edid_fb_mode_is_equal(use_aspect, mode, &mxc_cea_mode[i]))
-			break;
+			return i;
 	}
 
-	if (i == ARRAY_SIZE(mxc_cea_mode))
-		return 0;
-
-	return i;
+	return 0;
 }
 EXPORT_SYMBOL(mxc_edid_mode_to_vic);
 
diff --git a/include/video/mxc_edid.h b/include/video/mxc_edid.h
index 2feed16..1e89fd2 100644
--- a/include/video/mxc_edid.h
+++ b/include/video/mxc_edid.h
@@ -104,7 +104,6 @@ static inline unsigned long mxcPICOS2KHZ(u32 pixclock, u32 vmode) {
 			(999000999UL*4) : (1000000000UL*4)) / pixclock) + 1) >> 2;
 }
 
-int mxc_edid_var_to_vic(struct fb_var_screeninfo *var);
 int mxc_edid_mode_to_vic(const struct fb_videomode *mode);
 int mxc_edid_read(struct i2c_adapter *adp, unsigned short addr,
 	unsigned char *edid, struct mxc_edid_cfg *cfg, struct fb_info *fbi);
-- 
1.9.1

